### PR TITLE
feat: Custom hook api

### PR DESCRIPTION
**Pull request by timothyrusso**

### CODE REVIEW

File: src/pages/Home.js [hunk 1]
```diff
@@ -9,7 +9,8 @@
   =========================================================
   * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 */
-import { useState } from "react";
+import { useState, useEffect } from 'react';
+import { useApi } from '../api';
```
Comment (DannyVogel) [inline src/pages/Home.js:13]: I had to change the import reference as follows to make it work or it wouldn't compile:

import { useApi } from '../hooks/useApi';

File: src/pages/Home.js [hunk 2]
```diff
@@ -20,38 +21,31 @@ import {
...
+  const { Title } = Typography;
 
-  const onChange = (e) => console.log(`radio checked:${e.target.value}`);
+  const [coins, setCoins] = useState();
+  const { handleAllCoins } = useApi();
 
-  const [reverse, setReverse] = useState(false);
+  // useEffect(() => {
+  //   handleAllCoins().then((res) => setCoins(res));
+  // }, []);
```
Comment (DannyVogel) [inline src/pages/Home.js:48]: ESLINT says to remove the dependency array from useEffect if it is empty (probably just a stylistic thing)

File: src/pages/Home.js (hunk 1 above)
Comment (timothyrusso) [reply src/pages/Home.js]: Sorry I forgot to update the import, now it's correct

File: src/pages/Home.js (hunk 2 above)
Comment (timothyrusso) [reply src/pages/Home.js:48]: If we remove the dependency array the API will be called for an infinite time, using an empty array [] we are saying to run the useEffect only one time every render. It is a common ESLINT error

Comment (DannyVogel) [reply src/pages/Home.js:48]: ah true!

**Comment by DannyVogel** [comment]
ok to merge!

**Pull request merged**